### PR TITLE
Update MLflow UI

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Update MLflow UI.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Screen recording of manual QA (too large to attach on GitHub, let me know if you can't view it): 
https://drive.google.com/file/d/1EoilZo-LZLmjx5CzLWZBfNPRQ4roN4Jp/view?usp=sharing

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [x] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/server/js/src/common/sdk/PipelinesService.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { getJson } from '../utils/FetchUtils';
+
+export class PipelinesService {
```
Comment (harupy) [inline mlflow/server/js/src/common/sdk/PipelinesService.js:3]: This file is used in feature-store UI.

Comment (harupy) [reply mlflow/server/js/src/common/sdk/PipelinesService.js:3]: This class is used in feature-store UI. I'll update the copybara config to ignore this file.

File: mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js [hunk 2]
```diff
@@ -676,8 +691,20 @@ function SourceCellRenderer(props) {
 SourceCellRenderer.propTypes = { value: PropTypes.object };
 
 function VersionCellRenderer(props) {
-  const { version, name, type } = props.value;
-  return Utils.renderSourceVersion(version, name, type) || EMPTY_CELL_PLACEHOLDER;
+  // prettier-ignore
```
Comment (xanderwebs) [inline mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js:694]: This seems weird. Why are we ignoring prettier? cc @mohamad-arabi

**Review by xanderwebs — APPROVED** [review]
Left a few comments, looks generally good -- We really should do this more often 😄

File: mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js:694]: This function looks like in our private repo:

```js
function VersionCellRenderer(props) {
  // prettier-ignore
  const {
    // BEGIN-EDGE
    someProp,
    // END-EDGE
    version,
    name,
    type,
  } = props.value;
  return (
    // prettier-ignore
    Utils.renderSourceVersion(
      // BEGIN-EDGE
      someProp,
      // END-EDGE
      version,
      name,
      type,
    ) || EMPTY_CELL_PLACEHOLDER
  );
}
```

File: mlflow/server/js/src/experiment-tracking/components/ExperimentPage.test.js [hunk 3]
```diff
@@ -68,6 +68,11 @@ beforeEach(() => {
       search: '',
     },
   };
+
+  /* eslint-disable no-restricted-globals */
+  top.settings = {
+    autoMLEnabled: true,
+  };
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/ExperimentPage.test.js:75]: This looks like a edge block that should be excluded.

File: mlflow/server/js/src/common/utils/Utils.js [hunk 4]
```diff
@@ -852,13 +907,15 @@ class Utils {
     });
   }
 
-  static logErrorAndNotifyUser(e) {
+  static logErrorAndNotifyUser(e, passErrorToParentFrame = false, exceptionService = 'mlflow') {
```
Comment (harupy) [inline mlflow/server/js/src/common/utils/Utils.js:910]:
```suggestion
  static logErrorAndNotifyUser(e, passErrorToParentFrame = false) {
```

File: mlflow/server/js/src/experiment-tracking/components/ArtifactPage.js [hunk 5]
```diff
@@ -140,7 +156,10 @@ export class ArtifactPageImpl extends Component {
 
   render() {
     return (
-      <RequestStateWrapper requestIds={this.listArtifactRequestIds}>
+      <RequestStateWrapper
+        requestIds={this.listArtifactRequestIds}
+        // eslint-disable-next-line no-trailing-spaces
```
Comment (harupy) [inline mlflow/server/js/src/experiment-tracking/components/ArtifactPage.js:161]: @xanderwebs In our private repo, this part looks like this:

```suggestion
        // eslint-disable-next-line no-trailing-spaces
        description={SomeClass.VARIABLE_STORING_STRING}
```

Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/ArtifactPage.js:161]: I think we can remove this.

File: mlflow/server/js/src/experiment-tracking/components/ExperimentPage.test.js (hunk 3 above)
Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/ExperimentPage.test.js:75]: Removed.

File: mlflow/server/js/src/common/utils/Utils.js (hunk 4 above)
Comment (harupy) [reply mlflow/server/js/src/common/utils/Utils.js:910]: Removed.

File: mlflow/server/js/src/common/components/error-boundaries/AppErrorBoundary.js [hunk 6]
```diff
@@ -11,6 +11,7 @@ class AppErrorBoundary extends Component {
   }
 
   static propTypes = {
+    service: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/common/components/error-boundaries/AppErrorBoundary.js:14]: Removed :)

File: mlflow/server/js/src/common/components/error-boundaries/AppErrorBoundary.test.js [hunk 7]
```diff
@@ -8,7 +8,7 @@ describe('AppErrorBoundary', () => {
   let minimalProps;
 
   beforeEach(() => {
-    minimalProps = { children: 'testChild' };
+    minimalProps = { children: 'testChild', service: 'mlflow' };
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/common/components/error-boundaries/AppErrorBoundary.test.js:11]: Removed :)

File: mlflow/server/js/src/common/sdk/PipelinesService.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/common/sdk/PipelinesService.js:3]: Removed this file.

File: mlflow/server/js/src/common/utils/FetchUtils.js [hunk 8]
```diff
@@ -90,14 +90,13 @@ export const fetchEndpointRaw = ({
   timeoutMs = undefined,
 }) => {
   const url = getAjaxUrl(relativeUrl);
-  // eslint-disable-next-line prefer-const
-  let appName = 'mlflow';
 
+  // END-EDGE
```
[the remark this answers is not in the document]
Comment (harupy) [reply mlflow/server/js/src/common/utils/FetchUtils.js:94]: Looks like we forgot to remove `// BEGIN-EDGE`. Filed a PR in our internal repo to remove this unpaired edge comment.

File: mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js (hunk 2 above)
Comment (harupy) [reply mlflow/server/js/src/experiment-tracking/components/ExperimentRunsTableMultiColumnView2.js:694]: We might want to add a comment above `prettier-ignore` for clarity.

**Pull request merged**